### DISCUSSION TITLE
How do you pass null to a functions parameter using handlebars?

### DISCUSSION THREAD

**Discussion by preezie-yasir** [microsoft/semantic-kernel#6098] [Q&A] [answered]
When i use {{set "variable_name" null}} and pass variable_name as a parameter to a function call, it throws an error passing in i guess an empty string. Is there a way we can do this? Im using C#'s Semantic Kernel library Handlebars planners txt files

**Comment by sophialagerkranspandey** [comment]
Before proceeding, we'd love to understand the use case and why you're wanting to pass null? Thanks.

**Reply by preezie-yasir** [reply]
We're passing in an object to KernelArguments for the plan to work with. Some of the objects properties can be nullable and when being passed into a functions parameters within the plan, it crashes the plan.

**Reply by madsbolaris** [reply, marked as answer]
Thanks for raising this. I'm going to create an issue on this so that we can better investigate.

**Reply by madsbolaris** [reply]
My _initial_ thought is that Handlebars does provide functionality to [check for nulls](https://handlebarsjs.com/guide/builtin-helpers.html). Do you mind trying this to see if it works?

**Reply by preezie-yasir** [reply]
@matthewbolanos using conditional statements to check for nulls is fine but not when you're passing a variable that happens to be null into a function calls parameter.

**Reply by preezie-yasir** [reply]
I need to pass these variables into a function call because they need to be utilized in the case they are not null and i cant just wrap the whole function call in an if statement because i still need to call the function whether the variables are null or not
